zerocopy-derive/src/lib.rs

```diff
@@ -279,7 +316,11 @@ fn derive_known_layout_inner(ast: &DeriveInput, _top_level: Trait) -> Result<Tok
                 #repr
                 #[doc(hidden)]
                 #vis struct __ZerocopyKnownLayoutMaybeUninit<#params> (
-                    #(::zerocopy::util::macro_util::core_reexport::mem::MaybeUninit<#leading_fields_tys>,)*
+                    #(::zerocopy::util::macro_util::core_reexport::mem::MaybeUninit<
+                        <#ident #ty_generics as
+                            ::zerocopy::util::macro_util::Field<#leading_field_indices>
+                        >::Type
+                    >,)*
```

Comment:
Comment here explaining why we indirect via `Field` rather than just using the field types directly?

Reply:
[Done](https://github.com/google/zerocopy/compare/c7cfc8df03d86f6f93d73c8bc37729ad12eebd96..2cfcc5a8c3baf2a66dbb18fe633ba234a411a155#diff-977c29e6cf079bd7307f811df24f99633df603962785c583145eb6250b0e12daR318-R320)

Reply:
Also mention #2116 in this comment?

Reply:
[Done](https://github.com/google/zerocopy/compare/2cfcc5a8c3baf2a66dbb18fe633ba234a411a155..9f787224d7aa909c7c030422ded4f8495cacc35e)